shoop_tests/front/test_middleware.py

```diff
@@ -100,3 +100,25 @@ def test_intra_request_user_changing(rf, regular_user):
     assert request.user == AnonymousUser()
     assert request.person == shoop.core.models.AnonymousContact()
     assert request.customer == shoop.core.models.AnonymousContact()
+
+
+@pytest.mark.django_db
+def test_maintenance_mode(rf, regular_user, admin_user):
+    shop = shoop.core.models.Shop.objects.create(
+        name="test shop",
+        identifier="test",
+        status=shoop.core.models.ShopStatus.ENABLED,
+        maintenance_mode=True
+    )
+    mw = ShoopFrontMiddleware()
```

Comment:
Considering `ShoopFrontMiddleware` does `request.shop = Shop.objects.first()`, I don't see how the freshly created shop would end up there...

Maybe just modify `get_default_shop()` and return it to normal after the test?